src/server/game/Miscellaneous/Formulas.cpp

```diff
@@ -84,6 +84,13 @@ uint32 Acore::XP::Gain(Player* player, Unit* unit, bool isBattleGround /*= false
         }
 
         xpMod *= isBattleGround ? sWorld->getRate(RATE_XP_BG_KILL) : sWorld->getRate(RATE_XP_KILL);
+       
+        // if players dealt less than 50% of the damage and were credited anyway (due to CREATURE_FLAG_EXTRA_NO_PLAYER_DAMAGE_REQ), scale XP gained appropriately (linear scaling)
+        if (creature && creature->m_PlayerDamageReq)
+        {
+            xpMod *= 1.0f - 2.0f * creature->m_PlayerDamageReq / creature->GetMaxHealth();
+        }
+        
```

Comment:
@Krejza9 you got whitespaces on lines 87 and 93, that's why the codestyle check is failing in the CI

Reply:
done